storm-core/src/jvm/backtype/storm/utils/Utils.java

```diff
@@ -764,5 +778,30 @@ public static long zipFileSize(File myFile) throws IOException{
     public static double zeroIfNaNOrInf(double x) {
         return (Double.isNaN(x) || Double.isInfinite(x)) ? 0.0 : x;
     }
+
+    /**
+     * parses the arguments to extract jvm heap memory size.
+     * @param input
+     * @param defaultValue
+     * @return the value of the JVM heap memory setting in a java command.
+     */
+    public static Double parseWorkerChildOpts(String input, Double defaultValue) {
```

Comment:
I would prefer to have the name of this function indicate what it is parsing out of the childopts.